### ISSUE TITLE
not working

### ISSUE THREAD

**Issue by CHHE-ledtec** [alexbn71/AgentDVR-Telegram#6]
hi

 i downloded and added the sh file to the commands folder and created the folder.

activate motion detection and added the task 

execute command 

telegram_sendphotot

with parameters 

{ID} '{MSG}' {BASE64IMAGE}

but folder is empty

telegram bot is reciving/sending nothing

logs only tells me 

8:06:40 AM	Entre: LoadJson: getcameragrabs
8:06:40 AM	Entre: LoadJson: getcameragrabs
8:08:07 AM	Entre: LoadJson: getcameragrabs
8:08:07 AM	Entre: LoadJson: getcameragrabs
8:08:10 AM	Entre: LoadJson: getcameragrabs
8:08:10 AM	Entre: LoadJson: getcameragrabs
8:08:23 AM	Entre: LoadJson: getcameragrabs
8:08:23 AM	Entre: LoadJson: getcameragrabs
8:08:26 AM	Entre: LoadJson: getcameragrabs
8:08:26 AM	Entre: LoadJson: getcameragrabs

and thats it

 i am stuck soo need assistance

**Comment by CHHE-ledtec** [comment]
i am not getting any errors in the logs

**Comment by alexbn71** [comment]
Hi

to be sure that your detecor is set correctly, did you get alerts here?

![image](https://github.com/user-attachments/assets/6feb4e07-d47a-4525-922f-77fe8c14adb3)

If AgentDvr try to run the script you should see something like this in the log:

08:40:10	MyCamera: ProcessAlertEvent: Processing EXE (telegram_sendphoto.sh)
08:40:10	RunScript: /bin/bash -c "/AgentDVR/Commands/telegram_sendphoto.sh 28 'Manual Alert' >>>etc etc
08:40:12	RunScript: % Total % Received % Xferd Average Speed Time Time Time Current Dload Upload Total Spent Left Speed 0 0 0 0 0 0 0 0 >>>etc etc
08:40:12	RunScript: {"ok":true,"result":{"message_id" >>>etc etc

I think that your AgentDVR is not trying to execute the script since you should see at least some errors...

**Comment by CHHE-ledtec** [comment]
yes i am getting alert have over 100 allerts and omages in agentdvr

but as you say i dont see any action of exectuing the script .

testes a thng just now ..

the action was set to IF alert in zone 1 execute command

i changed it to IF Motion Detected in zone 1 execute command <--- and thiw works now i get execution of the script 

but getting 


7:18:33 AM	RunScript: /Agent/Commands/telegram_sendphoto.sh: line 15: /AgentDVR/Commands/grabs/grab64_3.jpg: No such file or directory /agent/Commands/telegram_sendphoto.sh: line 16: /AgentDVR/Commands/grabs/grab64d_3.jpg: No such file or directory curl: (26) Failed to open/read local data from file/application


i tried changing to lower case in the script but i am still getting 

RunScript: /agent/Commands/telegram_sendphoto.sh: line 15: /agentDVR/Commands/grabs/grab64_3.jpg: No such file or directory /agent/Commands/telegram_sendphoto.sh: line 16: /agentDVR/Commands/grabs/grab64d_3.jpg: No such file or directory curl: (26) Failed to open/read local data from file/application

Edit : found the issue in the .sh file the folder is named agentDVR when it only should be agent in the docker version.

**Comment by rrbb86** [comment]
Hello @alexbn71  thank you for this script. 

Unfortunately, the connection with Telegram doesn't work for me. No content arrives in the chat.

Do you have any idea what I should check?

agentDVR is used as a docker on unraid surver.

**Comment by alexbn71** [comment]
Hi,

are you sure that the event is triggered in agentdvr? Did you have the
captured screenshot visible inside alerts (top right)?

Il ven 18 ott 2024, 11:15 rrbb86 ***@***.***> ha scritto:

**Comment by MrDemonologist** [comment]
Hi! Please help
i receive next errors in log
it means "The specified file cannot be found"
i make some correction in telegram_sendphoto.sh
for example /Agent*RemoveDVR*/Commands/grabs/grab64_${ID}.jpg
because path to agent is C:\Program Files\Agent\Commands, not AgentDVR
and run again but it not helped
and grabs folder is empty
please check

19:54:56	Camera 2: ProcessAlertEvent: Processing EXE (telegram_sendphoto.sh)
19:54:56	RunScript: /bin/bash -c "C:\Program Files\Agent\Commands\telegram_sendphoto.sh 2 'Manual Alert' 
******************************blablabla"
19:54:56	ProcessAlertEvent: **Не удается найти указанный файл** at System.Diagnostics.Process.StartWithCreateProcess
(ProcessStartInfo startInfo) at CoreLogic.Utilities.Helper.RunScript(String filename, String args, Boolean noWindow) at 
CoreLogic.Objects.Helpers.AlertProcessor.<ProcessAlertEvent>d__8.MoveNext()
19:54:57	Camera 2: OpenWriter: StartSaving
19:54:57	Camera 2: .ctor: Generating timestamps for this recording
19:54:57	Open: OPEN RECORD
19:54:57	Open: written header
19:54:57	Camera 2: Open: Recording (Raw Writer CoreLogic.Sources.Combined.MediaStream)
19:54:57	Camera 2: RecorderRecordingOpened: Recording Opened
19:54:57	Camera 2: RecorderRecordingOpened: Recording Created

**Comment by electro-tek** [comment]
Looks like you're trying to run a shell script in Windows when you need a batch file equivalent.

**Comment by poisoh** [comment]
`00:21:45 Camera 3: ProcessAlertEvent: Processing EXE (telegram_sendphoto.sh)
00:21:45 RunScript: /bin/bash -c "/opt/AgentDVR/Commands/telegram_sendphoto.sh 3 'Manual Alert' /9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAA
..
b2RY+8aNx3aqNtzv4ufKg//2Q=="
00:21:45 RunScript: /opt/AgentDVR/Commands/telegram_sendphoto.sh: строка 15: /AgentDVR/Commands/grabs/grab64_3.jpg: Нет такого файла или каталога
/opt/AgentDVR/Commands/telegram_sendphoto.sh: строка 16: /AgentDVR/Commands/grabs/grab64d_3.jpg: Нет такого файла или каталога
curl: (26) Failed to open/read local data from file/application`

**Comment by Krolik2498** [comment]
Hi, thanks for your hard work. But it works for me every other time. Swears at long arguments. What could be the problem?

16:20:03 | RunScript: Executing: /bin/bash -c "/opt/AgentDVR/Commands/telegram.sh 1 ' /9j//gAQTGF2YzYxLjIyLjEwMAD/2wBDAAgODhAOEBMTExMTExYVFhcXFxYWFhYXFxcZGRkdHR0ZGRkXFxkZHBwdHSAhIB4eHR4hISMjIyoqKCgxMTI8PEj/xACfAAACAwEBAQEAAAAAAAAAAAAEBQIDBgEHAAgBAQEBAQEBAQAAAAAAAAAAAAABAgMEBQYQAAMAAgEEAQQBAwIDBQQGCwIBAxIRBBMhACIxFEEjMgVhUUIzcVKBJHK...............................
-- | --
16:20:03 | RunScript: Failed to execute process: An error occurred trying to start process '/bin/bash' with working directory '/opt/AgentDVR'. Argument list too long

the agent-dvr  is at 24 ubuntu, not in docker

**Comment by idstone** [comment]
Hi.
got the same error while using cameras HQ-Stream. I had to add  **--wrap=64** to the line with "base64 --decode".

My changed script look like this:

CHATID=xxxxxxx
TOKEN=xxxxxxxx
workdir=/tmp/grabs
#################################################
ID=$1
MSG=$2
BASE64IMAGE=$3

#ulimit -s was
#8192
ulimit -s 16384

#using /tmpfs and symlink
mkdir -p $workdir
cd $workdir
ln -s /tmp/grabs ./grabs

echo ${BASE64IMAGE} > ./grab64_${ID}.jpg
base64 --decode **--wrap=64** ./grab64_${ID}.jpg > ./grab64d_${ID}.jpg

curl -v -F chat_id=${CHATID} \
-F photo=@./grab64d_${ID}.jpg \
-F caption="${MSG}" \
https://api.telegram.org/bot${TOKEN}/sendPhoto

**Comment by idstone** [comment]
Hi!

First of all - BIG thx 2 alexbn71! :)

The base64-pic-output seems too long for sending via telegram-Bot
- with HQ-stream
- if action reacts on alert and
- "Object Recognition" is enabled.

If Object Recognition / AI Photos / Photos is enabled and AI Photos / CROP is active, pics are sent - but cropped.
Curiously a manual alert triggers correctly

I changed my setup to this:

#Install imagemagick
apt install imagemagick

#DISABLE
Object Recognition / AI Photos 

#CHANGE ALERT TO
{ID} '{NAME}'

```
grabs=/tmp/grabs
mkdir -p $grabs
if [ ! -d /opt/AgentDVR/Media/Commands/grabs ] ; then ln -s -T $grabs /opt/AgentDVR/Media/Commands/grabs ; fi

ID=$1
MSG=$2

curl -X 'GET' \
  'http://localhost:8090/photo.jpg?oid='${ID}'' \
  -H 'accept: image/jpeg' \
  --output $grabs/grab64dc_${ID}.jpg

# ... resize
convert -resize 30% $grabs/grab64dc_${ID}.jpg $grabs/grab64d_${ID}.jpg

# ...and send via telegram
curl -v -F chat_id=${CHATID} \
-F photo=@$grabs/grab64d_${ID}.jpg \
-F caption="${MSG}" \
https://api.telegram.org/bot${TOKEN}/sendPhoto

```

**Comment by alexbn71** [comment]
Hi @idstone, 

unfortunately I use docker and that tool should be integrated natively into the docker image :-(

**Comment by idstone** [comment]
Hey :)
resizing is just optional...
something seems to get wrong with the base64image, when the AI interpreted the pics and the action sends via "bash -c ..." from AgentDVR. A workmate tried to explain - but he failed  :D

Anyway - instead of using the base64, just "curl" an actual pic from the cam an let the bot send this... 
or am I understanding something wrong?

#DISABLE
Object Recognition / AI Photos

#CHANGE ALERT TO
{ID} '{NAME}'

and the script...

```
#!/bin/bash
## Alessandro Blason
## - Send a photo with caption
## version 0.0.2
#################################################

## UPDATE THESE VARIABLES 
CHATID=your telegram id
TOKEN=your bot token
#################################################

ID=$1
MSG=$2

curl -X 'GET' \
  'http://localhost:8090/photo.jpg?oid='${ID}'' \
  -H 'accept: image/jpeg' \
  --output /AgentDVR/Commands/grabs/grab64d_${ID}.jpg

curl -v -F chat_id=${CHATID} \
-F photo=@/AgentDVR/Commands/grabs/grab64d_${ID}.jpg \
-F caption="${MSG}" \
https://api.telegram.org/bot${TOKEN}/sendPhoto

```

**Comment by alexbn71** [comment]
Hi @idstone,

I don't like this solution... "curl" an actual pic .... On the other hand, this was my old solution because of the delay in elaboration, the capture of a photogram may result in a delay in respect of all the events that have been invoked.
Potentially you may lose the event in the image if this is very fast and it was near the edge of the camera.

**Comment by idstone** [comment]
Yeah, maybe it's a bit delayed... but I'm glad to receive notifications again.
However the issue with the base64 image seems to be a pretty deep-rooted problem.
I'd never seen some of the commands used today.
My colleague's comment was "no chance without disassembling the AgentDVR binary"  ;)

In any case, it seems certain that your script is perfectly fine. The problem lies elsewhere.

**Comment by alexbn71** [comment]
Hi @idstone,

did you see this?

<img width="809" height="407" alt="Image" src="https://github.com/user-attachments/assets/9fba7225-499a-4752-bd75-5775c410f42d" />

**Comment by idstone** [comment]
Hey again :)

No, I wasn't aware of that. But it's not listed in the actions either?!
Asap I got time, I'll try it out.

**Comment by idstone** [comment]
Hi again...
unfortunately it doesn't work with

{ID} '{NAME}' {BASE64IMAGE-NATIVE} in alert and telegram_sendphoto.sh

The return seems too long for sending via telegram-bot.

22:53:11	RunScript: Failed to execute process: An error occurred trying to start process '/bin/bash' with working directory '/opt/AgentDVR'. Argument list too long
